### ISSUE TITLE
Add "self" template module view helper

### ISSUE THREAD

**Issue by josh** [github/graphql-client#40]
``` erb
<%graphql
  fragment User on User {
    login
  }
%>
<% user = Views::Users::Show::User.new(user) %>

<%# Something like this should also work %>
<% user = self::User.new(user) %>
```

However, `self` is reserved, and so are a bunch of other good names. `view`, `template`, etc are already used by `ActionView::Base`.

##

CC: @nakajima

**Comment by nakajima** [comment]
Some ideas:

```rb
fragment::User.new(user)
graph::User.new(user)
graphql::User.new(user)
```

**Comment by josh** [comment]
So it could also match a query as well. The helper would be a stand in for the current view module. If the current template is `views/users/show/index.html.erb`, it'd be `"views/users/show". classify.constantize`

```
<%graphql
  fragment UserFragment on User {
    login
  }

  query UserQuery {
    ...UserFragment
  }
%>
<% user = self::UserFragment.new(user) %>
<% user = self::UserQuery.new(user) %>
```

**Comment by nakajima** [comment]
Could we get this working?

```rb
query::User.new(user)
fragment::User.new(user)
```

**Comment by josh** [comment]
> Could we get this working?

Yeah, I mean we can have an alias.

I guess I'm kinda hoping to put `_module` in the method since thats what it will be returning.

**Comment by josh** [comment]
For some reason I was thinking this would have to be implemented with an ivar, but it could just be a local injected statically into the compiled method. No stack tracking required.

Though, the current adapter only reimplements the `erb_implementation`. In order to get a handle on the original template path, we'd need a full `ActionView::Template::Handlers::ERB` subclass. Doable, just more of an invasive replacement and requires some API changes here.

Basically,

``` ruby
_module = Views::Users::Show
# rest of compiled template source
```

Just gets prepended to the compiled source.

**Comment by josh** [comment]
Likely superseded by magic solution https://github.com/github/graphql-client/pull/86
